### PR TITLE
Update `workerpool` to latest (avoiding deprecations)

### PR DESCRIPTION
**Pull request by nlfurniss**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,16 +38,16 @@
...
+    "broccoli-merge-trees": "^4.2.0",
     "broccoli-persistent-filter": "^3.0.0",
     "clone": "^2.1.2",
     "hash-for-dep": "^1.4.7",
     "heimdalljs": "^0.2.1",
     "heimdalljs-logger": "^0.1.9",
     "json-stable-stringify": "^1.0.1",
     "rsvp": "^4.8.4",
-    "workerpool": "^3.1.1"
+    "workerpool": "^6.0.2"
```
Comment (rwjblue) [inline package.json:50]: I think we have some changes needed here to accommodate this, from [their changelog]():

> * 5.0
>   * Deprecated option nodeWorker and created a new, more extensive option workerType giving full control over the selected type of worker. Added new option 'web' to enforce use a Web Worker. See #85, #74.
> * 6.0
>   * Removed deprecation warnings for options.nodeWorker (renamed to options.workerType) and pool.clear() (renamed to pool.terminate()).

See https://github.com/babel/broccoli-babel-transpiler/pull/180 for a reference...

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (nlfurniss) [reply package.json:50]: done!

File: package.json [hunk 2]
```diff
@@ -38,16 +38,16 @@
   "dependencies": {
     "@babel/core": "^7.11.0",
     "@babel/polyfill": "^7.10.4",
-    "broccoli-funnel": "^2.0.2",
-    "broccoli-merge-trees": "^3.0.2",
+    "broccoli-funnel": "^3.0.3",
+    "broccoli-merge-trees": "^4.2.0",
```
Comment (rwjblue) [inline package.json:42]: I'm pretty sure that these are completely unused at this point, can you confirm and remove?

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (nlfurniss) [reply package.json]: Done

**Review by rwjblue — APPROVED** [review]

**Pull request merged**